source/components/the-component-lifecycle.md

```diff
@@ -1,25 +1,129 @@
-Part of what makes components so useful is that they let you take
-complete control of a section of the DOM.  This allows for direct DOM
-manipulation, listening and responding to browser events, and using 3rd
-party JavaScript libraries in your Ember app.
+Part of what makes components so useful is that they let you take complete control of a section of the DOM.
+This allows for direct DOM manipulation, listening and responding to browser events, and using 3rd party JavaScript libraries in your Ember app.
 
-As components are rendered, re-rendered and finally removed, Ember
-provides _lifecycle hooks_ that allow you to run code at specific
-times in a component's life.
+As components are rendered, re-rendered and finally removed, Ember provides _lifecycle hooks_ that allow you to run code at specific times in a component's life.
 
-To get the most use out of a component, it is important to understand
-these lifecycle methods. The following hooks are a few of the most
-useful and commonly used in Ember apps.
+To get the most use out of a component, it is important to understand these lifecycle methods.
 
-## Attaching to the Component Element
+## Order of Lifecycle Hooks Called
+Listed below are the component lifecycle hooks in order of execution according to render scenario.
 
-Suppose you want to integrate your favorite date picker library into an
-Ember project. Typically, 3rd party JS/jQuery libraries require a DOM
-element to bind to. So, where is the best place to initialize and attach
-the library?
+### On Initial Render
 
-After a component successfully renders its backing HTML element into the
-DOM, it will trigger its [`didInsertElement()`][did-insert-element] hook.
+1. `didInitAttrs`
+2. [`didReceiveAttrs`](#toc_formatting-component-attributes-with-code-didreceiveattrs-code)
+3. `willRender`
+4. [`didInsertElement`](#toc_integrating-with-third-party-libraries-with-code-didinsertelement-code)
+5. [`didRender`](#toc_making-updates-to-the-rendered-dom-with-code-didrender-code)
+
+### On Re-Render 
+
+1. [`didUpdateAttrs`](#toc_resetting-presentation-state-on-attribute-change-with-code-didupdateattrs-code)
+2. [`didReceiveAttrs`](#toc_formatting-component-attributes-with-code-didreceiveattrs-code)
+3. `willUpdate`
+4. `willRender`
+5. `didUpdate`
+6. [`didRender`](#toc_making-updates-to-the-rendered-dom-with-code-didrender-code)
+
+#### Things That Trigger a Component Re-Render
```

Comment:
I think removing this section might clean things up some.  Maybe a better home would be API doc?